src/renderer/services/db/api/repositories/validator/attestations.ts

```diff
@@ -0,0 +1,56 @@
+import {IAttestationSearchOptions} from "@chainsafe/lodestar-validator/lib/db/interface";
+import {BulkRepository} from "../../repository";
+import {IDatabaseController, ISearchOptions} from "../../../../../../main/db/controller";
+import {Bucket, encodeKey} from "../../../schema";
+import {JSONSerializer} from "../../../serializers/json";
+import {types as mainnetTypes} from "@chainsafe/eth2.0-types/lib/ssz/presets/mainnet";
+import {Attestation, BLSPubkey} from "@chainsafe/eth2.0-types";
+
+export class ValidatorAttestationsRepository extends BulkRepository<Attestation> {
+    public constructor(db: IDatabaseController) {
+        super(db, JSONSerializer, Bucket.validatorAttestations, mainnetTypes.Attestation);
+    }
+
+    public async set(pubKey: BLSPubkey, attestation: Attestation): Promise<void> {
+        const key = this.getAttestationKey(pubKey, attestation);
+        await super.set(key, attestation);
+    }
+
+    public async deleteMany(pubKey: BLSPubkey, attestations: Attestation[]): Promise<void> {
+        const promises = [];
+        for (let i = 0; i < attestations.length; i++) {
+            const key = this.getAttestationKey(pubKey, attestations[i]);
+            promises.push(super.delete(key));
```

Comment:
I think there is method called batchDelete. I might be wrong

Reply:
Didn't notice it before but now I have tried it and it didn't work.. Tried several ways but for some reason items didn't get deleted although batch length was correct and no error was thrown.